<!--
   Author: Russell Toris, Worcester Polytechnic Institute
  Version: August 21, 2014
-->

<launch>
  <!-- AR Parameters -->
  <arg name="marker_size" default="8.0" />
  <arg name="bundle_files" default="$(find rail_ceiling)/bundles/coffee_table.xml" />
  <arg name="max_new_marker_error" default="0.08" />
  <arg name="max_track_error" default="0.2" />
  <arg name="node_name_prefix" default="ceiling_cam_tracker_" />
  <arg name="cam_image_topic_prefix" default="/ceiling_cam_" />
  <arg name="image_topic" default="/image_raw" />
  <arg name="cam_info_topic_prefix" default="$(arg cam_image_topic_prefix)" /> 
  <arg name="info_topic" default="/camera_info" />
  <arg name="output_frame_prefix" default="$(arg cam_image_topic_prefix)" />

  <!-- Assumes 5 Cameras -->
  <node pkg="ar_track_alvar" name="$(arg node_name_prefix)0" type="individualMarkersNoKinect" output="screen"
     args="$(arg marker_size)
           $(arg max_new_marker_error)
           $(arg max_track_error)
           $(arg cam_image_topic_prefix)0$(arg image_topic)
           $(arg cam_info_topic_prefix)0$(arg info_topic)
           $(arg output_frame_prefix)0
           $(arg bundle_files)">
    <remap from="visualization_marker" to="$(arg node_name_prefix)0/visualization_marker" />
    <remap from="ar_pose_marker" to="$(arg node_name_prefix)0/ar_pose_marker" />
  </node>
  <node pkg="ar_track_alvar" name="$(arg node_name_prefix)1" type="individualMarkersNoKinect" output="screen"
     args="$(arg marker_size)
           $(arg max_new_marker_error)
           $(arg max_track_error)
           $(arg cam_image_topic_prefix)1$(arg image_topic)
           $(arg cam_info_topic_prefix)1$(arg info_topic)
           $(arg output_frame_prefix)1
           $(arg bundle_files)">
    <remap from="visualization_marker" to="$(arg node_name_prefix)1/visualization_marker" />
    <remap from="ar_pose_marker" to="$(arg node_name_prefix)1/ar_pose_marker" />
  </node>
  <node pkg="ar_track_alvar" name="$(arg node_name_prefix)2" type="individualMarkersNoKinect" output="screen"
     args="$(arg marker_size)
           $(arg max_new_marker_error)
           $(arg max_track_error)
           $(arg cam_image_topic_prefix)2$(arg image_topic)
           $(arg cam_info_topic_prefix)2$(arg info_topic)
           $(arg output_frame_prefix)2
           $(arg bundle_files)">
    <remap from="visualization_marker" to="$(arg node_name_prefix)2/visualization_marker" />
    <remap from="ar_pose_marker" to="$(arg node_name_prefix)2/ar_pose_marker" />
  </node>
  <node pkg="ar_track_alvar" name="$(arg node_name_prefix)3" type="individualMarkersNoKinect" output="screen"
     args="$(arg marker_size)
           $(arg max_new_marker_error)
           $(arg max_track_error)
           $(arg cam_image_topic_prefix)3$(arg image_topic)
           $(arg cam_info_topic_prefix)3$(arg info_topic)
           $(arg output_frame_prefix)3
           $(arg bundle_files)">
    <remap from="visualization_marker" to="$(arg node_name_prefix)3/visualization_marker" />
    <remap from="ar_pose_marker" to="$(arg node_name_prefix)3/ar_pose_marker" />
  </node>
  <node pkg="ar_track_alvar" name="$(arg node_name_prefix)4" type="individualMarkersNoKinect" output="screen"
     args="$(arg marker_size)
           $(arg max_new_marker_error)
           $(arg max_track_error)
           $(arg cam_image_topic_prefix)4$(arg image_topic)
           $(arg cam_info_topic_prefix)4$(arg info_topic)
           $(arg output_frame_prefix)4
           $(arg bundle_files)">
    <remap from="visualization_marker" to="$(arg node_name_prefix)4/visualization_marker" />
    <remap from="ar_pose_marker" to="$(arg node_name_prefix)4/ar_pose_marker" />
  </node>
</launch>
